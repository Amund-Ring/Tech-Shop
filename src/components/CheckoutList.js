import { navigate } from "gatsby-link";
import React, { useEffect, useState } from "react";
import { getCartFromLocalStorage, removeFromCart } from "../data/dataHandler";
import * as styles from "../styles/CheckoutList.module.css";

function CheckoutList({ amountInCart, setAmountInCart }) {
  const [shoppingCart, setShoppingCart] = useState(getCartFromLocalStorage());

  useEffect(() => {
    setShoppingCart(getCartFromLocalStorage());
  }, [amountInCart]);


  return (
    <div className={styles.checkoutList}>
      <div className={styles.listContainer}>
        <div className={styles.itemLine}>
          <span className={styles.description}>Item</span>
          <span className={styles.amount}>Amount</span>
          <span className={styles.priceTitle}>Price</span>
          <span className={styles.bin}>Bin</span>
        </div>
        {shoppingCart.map((item, index) => {
          return (
            <div className={styles.itemLine} key={index}>
              <span
                onClick={() => {
                  navigate(`/product/?id=${item.productId}`);
                }}
                className={styles.description}
              >
                {`${item.name}
                 - ${item.color}
                ${item.storage ? " - " : ""}
                ${item.storage ? item.storage : ""}
                ${item.storage ? " GB" : ""}
                ${item.power ? " - " : ""}
                ${item.power ? item.power : ""}
                ${item.power ? " W" : ""}`}
              </span>
              <span className={styles.amount}>
                <div className={styles.amountAdjust}>
                  <span className={styles.adjustIncrDecr}>â€“</span>
                  <span className={styles.amountNr}>3</span>
                  <span className={styles.adjustIncrDecr}>+</span>
                </div>
              </span>
              <span className={styles.price}>{`${item.price},-`}</span>
              <span
                onClick={() => {
                  removeFromCart(item.lineItemId);
                  setAmountInCart(amountInCart - 1);
                }}
                className={styles.bin}
              >
                <svg
                  xmlns="http://www.w3.org/2000/svg"
                  width="16"
                  height="16"
                  viewBox="0 0 16 16"
                >
                  <path d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z" />
                  <path d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z" />
                </svg>
              </span>
            </div>
          );
        })}

        {/* <div className={styles.itemLine}></div>
        <div className={styles.itemLine}></div>
        <div className={styles.itemLine}></div>
        <div className={styles.itemLine}></div>
        <div className={styles.itemLine}></div>
        <div className={styles.itemLine}></div>
        <div className={styles.itemLine}></div>
        <div className={styles.itemLine}></div>
        <div className={styles.itemLine}></div> */}
      </div>
      <div className={styles.totalContainer}>
        <h3>Total: </h3>
      </div>
    </div>
  );
}

export default CheckoutList;
