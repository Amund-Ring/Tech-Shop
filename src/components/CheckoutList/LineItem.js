import React, { useEffect, useState } from "react";
import { navigate } from "gatsby-link";
import * as styles from "../../styles/CheckoutList.module.css";

import {
  removeLineFromCart,
  removeSingleItemFromCart,
  addAnotherItemLine,
  colorIsAvailable,
} from "../../data/dataHandler";

function LineItem({ item, amountInCart, setAmountInCart }) {
  const [plusButtonEnabled, setPlusButtonEnabled] = useState(
    colorIsAvailable(
      item[item.length - 1].productId,
      item[item.length - 1].color
    )
  );

  useEffect(() => {
    setPlusButtonEnabled(
      colorIsAvailable(
        item[item.length - 1].productId,
        item[item.length - 1].color
      )
    );
  }, [amountInCart]);

  return (
    <div className={styles.itemLine}>
      <span
        onClick={() => {
          navigate(`/product/?id=${item[item.length - 1].productId}`);
        }}
        className={styles.description}
        onKeyDown={() => {
          navigate(`/product/?id=${item[item.length - 1].productId}`);
        }}
        role="button"
        tabIndex={[]}
      >
        {`${item[item.length - 1].name}
                 - ${item[item.length - 1].color}
                ${item[item.length - 1].storage ? " - " : ""}
                ${
                  item[item.length - 1].storage
                    ? item[item.length - 1].storage
                    : ""
                }
                ${item[item.length - 1].storage ? " GB" : ""}
                ${item[item.length - 1].power ? " - " : ""}
                ${
                  item[item.length - 1].power ? item[item.length - 1].power : ""
                }
                ${item[item.length - 1].power ? " W" : ""}`}
      </span>
      <span className={styles.amount}>
        <div className={styles.amountAdjust}>
          <span
            className={styles.adjustIncrDecr}
            onClick={() => {
              setAmountInCart(
                removeSingleItemFromCart(item[item.length - 1].lineItemId)
              );
            }}
            onKeyDown={() => {
              setAmountInCart(
                removeSingleItemFromCart(item[item.length - 1].lineItemId)
              );
            }}
            role="button"
            tabIndex={[]}
          >
            â€“
          </span>
          <span className={styles.amountNr}>{item.length}</span>
          <span
            className={
              plusButtonEnabled ? styles.adjustIncrDecr : styles.plusDisabled
            }
            onClick={() => {
              setAmountInCart(addAnotherItemLine(item[item.length - 1]));
            }}
            onKeyDown={() => {
              setAmountInCart(addAnotherItemLine(item[item.length - 1]));
            }}
            role="button"
            tabIndex={[]}
          >
            +
          </span>
        </div>
      </span>
      <span className={styles.price}>
        <span>{`${item[item.length - 1].price},-`}</span>
      </span>
      <span
        onClick={() => {
          setAmountInCart(
            removeLineFromCart(
              item[item.length - 1].productId,
              item[item.length - 1].color
            )
          );
        }}
        className={styles.bin}
        onKeyDown={() => {
          setAmountInCart(
            removeLineFromCart(
              item[item.length - 1].productId,
              item[item.length - 1].color
            )
          );
        }}
        role="button"
        tabIndex={[]}
      >
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="16"
          height="16"
          viewBox="0 0 16 16"
        >
          <path d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z" />
          <path d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z" />
        </svg>
      </span>
    </div>
  );
}

export default LineItem;
