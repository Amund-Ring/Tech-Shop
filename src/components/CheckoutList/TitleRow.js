import React from "react";
import * as styles from "../../styles/CheckoutList.module.css";


function TitleRow() {
  return (
    <div className={styles.itemLine}>
      <span className={`${styles.description} ${styles.descriptionTitle}`}>
        Item
      </span>
      <span className={styles.amount}>Amount</span>
      <span className={styles.priceTitle}>Price</span>
      <span className={`${styles.bin} ${styles.binTitle}`}>Bin</span>
    </div>
  );
}

export default TitleRow;
