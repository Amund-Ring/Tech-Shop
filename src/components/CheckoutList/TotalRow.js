import React from "react";
import * as styles from "../../styles/CheckoutList.module.css";

function TotalRow({ total }) {
  return (
    <div className={styles.totalContainer}>
      <span></span>
      <h3>
        Total: <span>{Number(total).toLocaleString("no")},-</span>
      </h3>
    </div>
  );
}

export default TotalRow;
