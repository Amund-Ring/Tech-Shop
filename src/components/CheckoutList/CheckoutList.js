import React, { useEffect, useState } from "react";
import * as styles from "../../styles/CheckoutList.module.css";
import { getSortedCart, getTotalSum } from "../../data/dataHandler";

import LineItem from "./LineItem";
import TitleRow from "./TitleRow";
import TotalRow from "./TotalRow";

function CheckoutList({ amountInCart, setAmountInCart }) {
  const [sortedCart, setSortedCart] = useState(getSortedCart());
  const [total, setTotal] = useState(getTotalSum());

  useEffect(() => {
    setSortedCart(getSortedCart());
    setTotal(getTotalSum());
  }, [amountInCart]);

  return (
    <div className={styles.checkoutList}>
      <div className={styles.listContainer}>
        <TitleRow />
        {sortedCart.map((item, index) => {
          return (
            <LineItem
              item={item}
              key={index}
              amountInCart={amountInCart}
              setAmountInCart={setAmountInCart}
            />
          );
        })}
      </div>
      <TotalRow total={total} />
    </div>
  );
}

export default CheckoutList;
