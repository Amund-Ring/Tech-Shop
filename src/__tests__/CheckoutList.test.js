import React from "react";
import renderer from "react-test-renderer";
import CheckoutList from "../components/CheckoutList/CheckoutList";

describe("CheckoutList", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<CheckoutList amountInCart={6} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
