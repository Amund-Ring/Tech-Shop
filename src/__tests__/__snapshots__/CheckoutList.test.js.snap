// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutList renders correctly 1`] = `
<div>
  <div>
    <div>
      <span
        className="undefined undefined"
      >
        Item
      </span>
      <span>
        Amount
      </span>
      <span>
        Price
      </span>
      <span
        className="undefined undefined"
      >
        Bin
      </span>
    </div>
  </div>
  <div>
    <span />
    <h3>
      Total: 
      <span>
        0
        ,-
      </span>
    </h3>
  </div>
</div>
`;
